core-skills/web-application-development-with-react/README.md

```diff
@@ -0,0 +1,74 @@
+# Web Application Development with React
+
+## Summary
+
+1. Can describe situations where you should/shouldn't decide to use React for a project
+2. Can setup a React project from scratch in a repeatable way
+3. Can describe the different parts of the Javascript ecosystem (e.g. NPM, Webpack, etc)
+4. Can demonstrate using TDD on business logic within a React code base
+5. Can demonstrate strategies to achieve clear separation of business logic and display logic
+with React components
+6. Can demonstrate applying sensible styling to React components
+7. Can demonstrate strategies and tools to run tests against an API within a React application
+
+## Sections
+
+1. [React - Why and when should I use it?](./why-and-when.md)
+2. [Setting up a React project](./setup.md)
+3. [React Fundamentals](./fundamentals/README.md)
+4. [Prototyping React Components](./prototyping.md)
+5. [Styling React Components](./prototyping.md)
+
+## Mark Scheme
+
+### Hedgehog
+
+Learning the fundamentals - How not to hurt yourself with React
+
+- [ ] Can demonstrate an understanding of the benefits of using React
+- [ ] Creates an application with the following:
+  - The application can be run on any machine:
+    - [ ] The application has `make` recipes for `serve` and `test` which
+      only depend on the host machine having docker installed
+  - Setup/Tooling
+    - The candidate has setup the project with the following tools:
+      - [ ] A sandbox environment to quickly prototype components within (e.g. Storybook)
+      - [ ] Hot-reloading
+      - [ ] Tests (The ability to write them, writing one is not a requirement at this stage)
+  - Components
+    - The candidate is able to explain what the following are:
+      - [ ] A component
+      - [ ] The use of props within components
+      - [ ] The use of state within components
+      - [ ] The use of children within components
+      - [ ] Responding to user events
+- Demonstrating understanding of React fundamentals
+  - The candidate can explain the following:
+      - [ ] How to style components
```

Comment:
Style in any possible way? What's the criteria here?

Is `style={{ color: 'blue' }}` as good as hooking up emotion, react css modules, etc...?

Reply:
Ha yeah I read that and immediately thought of the `style` prop.  If you aren't wanting me to use the style prop you should call that out.

Reply:
I had assumed style prop would be sufficient for Hedgehog, and using 'proper' methods comes under architecture in later levels